Suppress output from mail parser warnings

skipkayhil opened the pull request:
### Motivation / Background

Previously, requiring any of the ragel generated parsers in mail would output tons of warnings in tests, making output much harder to read (especially in Railties).

### Detail

This commit extends the RaiseWarnings patch to suppress output from these mail parsers.

The suppression can be removed once mikel/mail#1557 or mikel/mail#1551 or any other PR fixes the issue

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* ~[ ] Tests are added or updated if you fix a bug or add a feature.~
* ~[ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.~

skipkayhil committed:
Suppress output from mail parser warnings

Previously, requiring any of the ragel generated parsers in mail would
output tons of warnings in tests, making output much harder to read
(especially in Railties).

This commit extends the RaiseWarnings patch to suppress output from
these mail parsers.

The suppression can be removed once mikel/mail#1557 or mikel/mail#1551
or any other PR fixes the issue

byroot reviewed (approved):
👍

byroot commented on activesupport/lib/active_support/testing/strict_warnings.rb:
```diff
@@ -12,7 +12,16 @@ module RaiseWarnings
+
+  SUPPRESSED_WARNINGS = Regexp.union(
+    # TODO: remove if https://github.com/mikel/mail/pull/1557 or similar fix
+    %r{/lib/mail/parsers/.*statement not reached},
```
Hum, doesn't seem to work, maybe it's loaded too late? https://buildkite.com/rails/rails/builds/94131#01868059-d70b-47ae-8cd2-3b6a1fa78824/1035-1038

byroot replied:
Hum, right it seem to fix actionmailer, but not actionmailbox.

skipkayhil replied:
[ba028a4](https://github.com/rails/rails/pull/47484/commits/ba028a48f6d1febf01a74058b753b340188eb6ef) should fix it, I can squash if you think they should go together

byroot replied:
Nah it's ok

skipkayhil committed:
Fix Action Mailbox, Action Text strict warnings

These appear to have been missed when the change was made in
d917896f45ad680f43036d8ff6d48d7e5e198d23

skipkayhil commented:
I missed at least one more:

https://buildkite.com/rails/rails/builds/94132#0186806f-8a23-4d07-ae8f-3b937f8517ae

I'll fix that and then verify all of railties is good
